tsa-core/src/main/kotlin/org/usvm/machine/interpreter/inputdict/Modification.kt

```diff
@@ -0,0 +1,62 @@
+package org.usvm.machine.interpreter.inputdict
+
+import org.usvm.UBoolExpr
+import org.usvm.machine.TvmContext
+import org.usvm.machine.splitHeadTail
+import kotlin.collections.plus
+
+/**
+ * Represent the single modification over a set of dictionary keys.
+ * Whenever you encounter `l: List<Modification>`, assume that `l.first()`
+ * is the last modification applied (and `l.last()` is the first modification applied).
+ */
+sealed interface Modification {
+    data class Remove(
+        val key: KeyType,
+    ) : Modification
+
+    data class Store(
+        val key: KeyType,
+        val guard: UBoolExpr,
+    ) : Modification
+}
+
+/** @return the condition that would be of a corresponding key in `getKeys` */
+fun List<Modification>.createKeyCondition(
+    ctx: TvmContext,
+    t: KExtended,
+): UBoolExpr {
+    val (head, tail) = splitHeadTail() ?: return ctx.trueExpr
+    val prevCond = tail.createKeyCondition(ctx, t)
+    return when (head) {
+        is Modification.Store -> with(ctx) { prevCond or (t eq head.key.toExtendedKey(ctx)) }
+        is Modification.Remove -> with(ctx) { prevCond and (t neq head.key.toExtendedKey(ctx)) }
+    }
```

Comment:
Is it true that if all modifications are `store`, then the result condition is always `trueExpr`? If so, what is supposed to mean?